zulipterminal/ui_tools/views.py

```diff
@@ -1168,7 +1168,32 @@ def __init__(self, controller: Any, title: str) -> None:
 
         super().__init__(controller, widgets, "HELP", popup_width, title)
 
+class FileUploadView(PopUpView):
+    def __init__(
+        self,
+        controller: Any,
+        title: str,
+    ) -> None:
+        self.controller = controller
+        self.model = controller.model
+        self.uri = None
+        max_cols, max_rows = controller.maximum_popup_dimensions()
+        self.predefined_text = urwid.Text("Location : ")
+        self.file_location_edit = urwid.Edit()
+        columns = [self.predefined_text,self.file_location_edit]
+        # body_list = [self.file_location_edit]
+        super().__init__(
+            controller,
+            columns,
+            "FILE_UPLOAD",
+            max_cols,
+            title,
+            # urwid.Pile(msg_box.header),
+            # urwid.Pile(msg_box.footer),
+        )
 
+    def keypress(self, size: urwid_Size, key: str) -> str:
+        return super().keypress(size, key)
```

Comment:
Maybe execute a function/some logic when the user hits `ENTER`

Reply:
@mounilKshah  The `FileUploadView` is derived from `PopUpView`. `PopUpView` is supposed to get closed on pressing the same key which opened. So I think it is natural that the user wants to close as soon as he has uploaded the file.
As Neil is suggesting considering using something different [(relevant chat)](https://chat.zulip.org/#narrow/stream/206-zulip-terminal/topic/File.20Upload.20.23T529.20.23T1414/near/1607421) than `PopUpView`, we should first discuss a bit about the Base classes to inherit from and then come back to chose appropriate keys and their actions.